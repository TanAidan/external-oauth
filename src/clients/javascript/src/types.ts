declare namespace Types {
  export interface ConnectOptions {
    authId?: string
    setupId?: string
    params?: any
  }

  export interface ConnectSuccess {
    authId: string
  }

  export interface ConnectError extends Error {}

  export interface IntegrationOptions {
    authId?: string
    setupId?: string
  }

  export type RequestHeaders = Record<string, string | number | undefined>
  export type RequestMethod = 'GET' | 'DELETE' | 'HEAD' | 'OPTIONS' | 'POST' | 'PUT' | 'PATCH'

  export type RequestQueryString = Record<string, string | number>
  export interface RequestParameters {
    headers?: RequestHeaders
    query?: RequestQueryString
    body?: any
  }
}

export default Types
