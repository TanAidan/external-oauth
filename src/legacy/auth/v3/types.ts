import { Request } from 'express'
import Integration from '../../functions/integration'
import Knex from 'knex'

export interface IAuthContext {
  clientId: string
  connectParams: any
  buid?: string
  setupId: string
  isCallback?: boolean
  store: Knex
  configuration: SetupRecord
}

interface SetupRecord {
  setup_id: string
  scopes: string[] | undefined
  buid: string
  configuration: any
}

interface IContextSessionData {
  session: {
    context: any
  }
}

export enum EDashboardApiErrorCodes {
  ENVIRONMENT_NOT_FOUND = 'ENVIRONMENT_NOT_FOUND',
  INTEGRATION_NOT_FOUND = 'INTEGRATION_NOT_FOUND'
}

export interface IDashboardApiResponse {
  errors: [
    {
      code: EDashboardApiErrorCodes
      message: string
    }
  ]
  data: {
    viewer: {
      canInvoke: boolean
      hasAccessTo: boolean
      clientId: string
      setupId: string
      buid: string
    }
  }
}

export type TConnectContextRequest = Request &
  IContextSessionData &
  IAuthContext & {
    params: {
      buid: string
    }
    query: {
      clientId: string
      setupId: string
    }
    integration?: Integration
    fullLogs?: boolean
  }

export type TCallbackContextRequest = Request & IContextSessionData & IAuthContext

export enum ELocalRedirectMethod {
  Inline = 'inline',
  Localhost = 'localhost'
}

type TNameValues = Record<string, any>

export type TIntegrationConfig = TNameValues & {
  authType: EAuthType
  config?: {
    scope?: string[]
    state?: any
  }
}

export interface IAuthConfig {
  integrationConfig: TIntegrationConfig
  setupDetails: TNameValues
}

export enum EAuthType {
  NoAuth = 'NO_AUTH',
  OAuth1 = 'OAUTH1',
  OAuth2 = 'OAUTH2'
}

interface IAuthConfigSessionData {
  session: {
    authConfig: IAuthConfig
  }
}

export type TConnectConfigRequest = Request & IAuthContext & IAuthConfig & IAuthConfigSessionData

export type TCallbackConfigRequest = Request & IAuthConfig & IAuthConfigSessionData

export interface IAuthId {
  authId: string
}

export type TAuthIdRequest = Request &
  IAuthContext &
  IAuthId & {
    session: {
      authId: string
      context: {
        setupId: string
      }
    }
  }

export interface IOAuth1Credentials {
  accessToken: string
  tokenSecret: string
  consumerKey: string
  consumerSecret: string
  expiresIn: number
}

export interface IOAuth2Credentials {
  accessToken: string
  refreshToken: string
  idToken: string
  idTokenJwt?: any
  expiresIn: number
}

export interface IWithAuthCredentials {
  credentials: IOAuth1Credentials | IOAuth2Credentials
  tokenResponse?: OAuthTokenResponse
}

export type TAuthenticateRequest = Request & IAuthContext & IAuthConfig & IWithAuthCredentials

export type AuthSuccessRequest = Request &
  IAuthContext &
  IAuthId &
  IWithAuthCredentials & { buid: string; integrationConfig?: TIntegrationConfig } & { store: Knex }

export type TRevokeRequest = Request & { buid: string }

export type TErrorHandlerRequest = Request & Partial<IAuthContext> & Partial<IAuthId>

export interface IFetchAuthDetailsParams {
  buid: string
  scopedUserDataTableName: string
  servicesTableName: string
  integration: Integration
  environmentIdentifier: string
  authId?: string
  setupId?: string
  configuration: any
  setupIdFromRequest?: boolean
  store: any
}

export interface BasicAuthDetails {
  username: string
  password: string
}

export interface ApiKeyAuthDetails {
  apiKey: string
}

export enum OAuth1SignatureMethod {
  HmacSha1 = 'HMAC-SHA1',
  RsaSha1 = 'RSA-SHA1',
  PlainText = 'PLAINTEXT'
}

interface OAuthAuthDetails {
  callbackParams: Record<string, string>
  tokenResponse?: OAuthTokenResponse
  updatedAt: number
}

export interface OAuth1AuthDetails extends OAuthAuthDetails {
  accessToken: string
  expiresIn: number
  tokenSecret: string
  consumerKey: string
  consumerSecret: string
  signatureMethod: 'HMAC-SHA1' | 'PLAINTEXT' | 'RSA-SHA1'
}

export interface OAuth2AuthDetails extends OAuthAuthDetails {
  accessToken: string
  callbackParams: Record<string, string>
  clientId: string
  clientSecret: string
  expiresIn: number
  idToken?: string
  idTokenJwt?: any
  refreshToken?: string
}

export interface OAuthTokenResponse {
  body: any
  headers: Record<string, string>
}

export type AuthDetails = BasicAuthDetails | ApiKeyAuthDetails | OAuth1AuthDetails | OAuth2AuthDetails | undefined
