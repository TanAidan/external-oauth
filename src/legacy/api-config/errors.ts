import { UserError } from '../errors'

export class InvalidApiConfig extends UserError {
  constructor(label: string, subAttribute: string | undefined, variableName: string) {
    const subText = subAttribute ? ` '${subAttribute}'` : ''
    super(
      `The API's ${label}${subText} has been configured with an invalid variable \${${variableName}}.`,
      422,
      'INVALID_API_CONFIG'
    )
  }
}

export class MissingApiConfigHeader extends UserError {
  constructor(label: string, subAttribute: string | undefined, variableName: string) {
    const subText = subAttribute ? ` '${subAttribute}'` : ''
    const header = variableName.split('.')[1]
    super(
      `The API's ${label}${subText} configuration requires the '${header}' header to be passed with each request.`,
      400,
      'MISSING_API_CONFIG_HEADER'
    )
  }
}

export class MissingApiConfigConnectParam extends UserError {
  constructor(label: string, subAttribute: string | undefined, variableName: string) {
    const subText = subAttribute ? ` '${subAttribute}'` : ''
    const param = variableName.split('.')[1]
    super(
      // tslint:disable-next-line:max-line-length
      `The API's ${label}${subText} configuration requires the '${param}' query parameter to be passed when connecting.`,
      400,
      'MISSING_API_CONFIG_HEADER'
    )
  }
}
