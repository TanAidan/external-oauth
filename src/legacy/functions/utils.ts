// Is the given object empty e.g. {}?
export function isEmpty(obj: Object) {
  return Object.getOwnPropertyNames(obj).length === 0
}
