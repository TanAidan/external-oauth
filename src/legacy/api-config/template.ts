import at from 'lodash.at'

import { CustomError } from '../errors'

// Matches escaped characters eg. "\$" or variables with optional surrounding whitespace eg. "${  name  }"
const parserRegex = /\\.|\$\{\s*([^}\s]*)\s*\}/g

const expandString = (template: string, templatePath: string, variables: any) =>
  template.replace(parserRegex, (match: string, variableName: string) => {
    if (match[0] === '\\') {
      return match[1]
    }

    const value = at<string | undefined>(variables, variableName)[0]

    if (value === undefined) {
      throw new UndefinedVariable(templatePath, variableName)
    }

    return value
  })

const expandArray = (template: any, templatePath: string, variables: any) =>
  template.map((element, i: number) => expand(element, `${templatePath}[${i}]`, variables))

const expandObject = (template: any, templatePath: string, variables: any) => {
  const result = {}

  for (const [attributeName, childTemplate] of Object.entries(template)) {
    const childPath = templatePath ? `${templatePath}.${attributeName}` : attributeName

    result[attributeName] = expand(childTemplate, childPath, variables)
  }

  return result
}

const expand = (template: any, templatePath: string, variables: any) => {
  if (Array.isArray(template)) {
    return expandArray(template, templatePath, variables)
  }

  if (typeof template === 'string') {
    return expandString(template, templatePath, variables)
  }

  if (typeof template === 'number') {
    return template
  }

  return expandObject(template, templatePath, variables)
}

/**
 * Performs variable interpolation on strings. The strings may be contained within
 * nested objects or arrays. Numbers are returned unchanged.
 *
 * Examples variables `{ name: 'Bearer', address: { city: 'Paris' } }`:
 *
 * Example cases:
 *
 *   `{ greeting: 'Hi ${name}', count: 5 }` => `{ greeting: 'Hi Bearer', count: 5 }`
 *
 *   `['City is ${address.city}', { busy: true }]` => `['City is Paris', { busy: true }]`
 *
 *   `'Hey there "${name}"'` => `'Hey there "Bearer"'`
 *
 *   `'Not a variable: \\${name}'` => `'Not a variable: ${name}'`
 *
 *   `{ a: 'Oops ${some.age}' }` => throws `UndefinedVariable(templatePath: 'a', variableName: 'some.age')`
 */
export const expandTemplate = (template: any, variables: any) => expand(template, '', variables)

export class UndefinedVariable extends CustomError {
  constructor(public templatePath: string, public variableName: string) {
    super(`Undefined variable ${variableName} used in template at ${templatePath}`)
  }
}
