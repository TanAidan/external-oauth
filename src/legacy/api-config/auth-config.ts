import pick from 'lodash.pick'

import { expandTemplate, UndefinedVariable } from './template'
import { EAuthType, OAuth1SignatureMethod } from '../auth/v3/types'
import { InvalidApiConfig, MissingApiConfigConnectParam } from './errors'
import { AuthorizationMethod, BodyFormat } from '../auth/clients/oauth2'
import { GrantType } from '../auth/v3/strategies/oauth2'

const labels: Record<string, string> = {
  accessTokenURL: 'Access Token URL',
  authorizationURL: 'Authorization URL',
  authorizationParams: 'Authorization Param',
  requestTokenURL: 'Request Token URL',
  tokenParams: 'Token Param',
  tokenURL: 'Token URL',
  refreshURL: 'Refresh URL',
  userAuthorizationURL: 'User Authorization URL'
}

export const expandAuthConfig = ({ connectParams, authConfig }: ExpandAuthConfigParams): AuthConfig => {
  const template = pick(
    authConfig,
    'accessTokenURL',
    'authorizationURL',
    'authorizationParams',
    'requestTokenURL',
    'tokenParams',
    'tokenURL',
    'refreshURL',
    'userAuthorizationURL'
  )

  try {
    return {
      ...authConfig,
      ...expandTemplate(template, { connectParams })
    }
  } catch (e) {
    if (!(e instanceof UndefinedVariable)) {
      throw e
    }

    const { templatePath, variableName } = e
    const [attributeName, subAttributeName] = templatePath.split('.')
    const label = labels[attributeName]

    if (e.variableName.startsWith('connectParams.')) {
      throw new MissingApiConfigConnectParam(label, subAttributeName, variableName)
    }

    throw new InvalidApiConfig(label, subAttributeName, variableName)
  }
}

interface ExpandAuthConfigParams {
  connectParams: Record<string, string>
  authConfig: AuthConfig
}

export interface OAuth1AuthConfig {
  authType: EAuthType.OAuth1
  accessTokenURL: string
  authorizationParams?: Record<string, string>
  requestTokenURL: string
  signatureMethod?: OAuth1SignatureMethod
  tokenParams?: Record<string, string>
  userAuthorizationURL: string
}

interface OAuth2AuthConfig {
  authType: EAuthType.OAuth2
  authorizationMethod?: AuthorizationMethod
  bodyFormat?: BodyFormat
  config?: {
    scope?: string[]
  }
  tokenParams?: Record<string, string>
  tokenURL: string
  refreshURL?: string
}

export interface OAuth2AuthCodeConfig extends OAuth2AuthConfig {
  grantType: GrantType.AuthCode
  authorizationURL: string
  authorizationParams?: Record<string, string>
}

export interface OAuth2ClientCredentialsConfig extends OAuth2AuthConfig {
  grantType: GrantType.ClientCredentials
}

export type AuthConfig = OAuth1AuthConfig | OAuth2AuthConfig
