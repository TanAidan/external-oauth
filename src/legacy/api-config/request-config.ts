import merge from 'merge'
import { OAuth } from 'oauth'

import { expandTemplate, UndefinedVariable } from './template'
import { EAuthType, AuthDetails, OAuth1AuthDetails } from '../auth/v3/types'
import { MissingApiConfigHeader, InvalidApiConfig, MissingApiConfigConnectParam } from './errors'

const labels: Record<keyof RequestConfig, string> = {
  baseURL: 'Request Base URL',
  headers: 'Request Header',
  params: 'Request Param'
}

export const expandRequestConfig = (params: ExpandRequestConfigParams): RequestConfig => {
  try {
    return expandTemplate(params.requestConfig, getVariables(params))
  } catch (e) {
    if (!(e instanceof UndefinedVariable)) {
      throw e
    }

    const { templatePath, variableName } = e
    const [attributeName, subAttributeName] = templatePath.split('.')
    const label = labels[attributeName]

    if (e.variableName.startsWith('connectParams.')) {
      throw new MissingApiConfigConnectParam(label, subAttributeName, variableName)
    }

    if (e.variableName.startsWith('headers.')) {
      throw new MissingApiConfigHeader(label, subAttributeName, variableName)
    }

    throw new InvalidApiConfig(label, subAttributeName, variableName)
  }
}

const tryExpandBaseURL = (baseURL: string, variables: any) => {
  try {
    return expandTemplate(baseURL, variables)
  } catch (e) {
    return baseURL
  }
}

const getVariables = ({
  auth,
  authType,
  connectParams,
  headers,
  method,
  path,
  requestConfig
}: ExpandRequestConfigParams) => {
  const partialVariables = { auth, connectParams, headers }
  // We want any expansion errors to be caught later in the process when
  // we have the proper context for error messages
  const baseURL = tryExpandBaseURL(requestConfig.baseURL, partialVariables)

  return merge.recursive(true, partialVariables, {
    auth: additionalAuthVariables({ auth, authType, baseURL, method, path })
  })
}

const additionalAuthVariables = ({ auth, authType, baseURL, method, path }: AdditionalAuthVariablesParams) => {
  switch (authType) {
    case EAuthType.OAuth1:
      return { oauth1: getOAuth1Credentials({ baseURL, method, path, auth: auth as OAuth1AuthDetails }) }
    default:
      return {}
  }
}

export const getOAuth1Credentials = ({ baseURL, method, path, auth }: IGetOAuth1CredentialsParams) => {
  const { consumerKey, consumerSecret, accessToken, tokenSecret, signatureMethod } = auth

  const absUrl = `${baseURL.replace(/\/+$/, '')}/${path.replace(/^\/+/, '')}`

  const oauth = new OAuth(
    '', // requestUrl
    '', // accessUrl
    consumerKey,
    consumerSecret,
    '1.0', // version
    '', // authorize_callback
    signatureMethod
  )

  return oauth.authHeader(absUrl, accessToken, tokenSecret, method).replace(/^OAuth /, '')
}

interface ExpandRequestConfigParams {
  auth: AuthDetails
  authType: EAuthType
  connectParams: Record<string, string>
  headers: Record<string, string>
  method: string
  path: string
  requestConfig: RequestConfig
}

export interface RequestConfig {
  baseURL: string
  headers: Record<string, string>
  params: Record<string, string>
}

interface AdditionalAuthVariablesParams {
  auth: AuthDetails
  authType: EAuthType
  baseURL: string
  method: string
  path: string
}

interface IGetOAuth1CredentialsParams {
  auth: OAuth1AuthDetails
  baseURL: string
  method: string
  path: string
}
