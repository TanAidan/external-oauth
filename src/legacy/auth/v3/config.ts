import { asyncMiddleware } from '../../errorHandler'
import { IAuthConfig, EAuthType, TConnectConfigRequest, TCallbackConfigRequest } from './types'
import { NextFunction, Response } from 'express'
import {
  getConfig
  // getSetupDetails
} from '../clients/integrations'
import { InvalidAuthType } from './errors'
import { expandAuthConfig } from '../../api-config/auth-config'

const getAuthConfig = async (req: TConnectConfigRequest) => {
  const { connectParams, buid, configuration } = req

  return {
    setupDetails: configuration,
    integrationConfig: expandAuthConfig({
      connectParams,
      authConfig: (await getConfig({
        buid: buid!
      })) as any
    })
  } as IAuthConfig
}

const copyConfig = (src: IAuthConfig, dest: IAuthConfig) => {
  dest.integrationConfig = src.integrationConfig
  dest.setupDetails = src.setupDetails
}

export const connectConfig = asyncMiddleware(async (req: TConnectConfigRequest, _res: Response, next: NextFunction) => {
  const authConfig = await getAuthConfig(req)
  const { authType } = authConfig.integrationConfig

  if (!Object.values(EAuthType).includes(authType)) {
    throw new InvalidAuthType(authType)
  }

  copyConfig(authConfig, req)
  req.session.authConfig = authConfig

  next()
})

export const callbackConfig = (req: TCallbackConfigRequest, _res: Response, next: NextFunction) => {
  copyConfig(req.session.authConfig, req)

  next()
}
