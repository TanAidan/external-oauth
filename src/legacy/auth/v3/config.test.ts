import { MiddlewareTestHarness, session } from '../../../../tests/utils'
import { connectConfig, callbackConfig } from './config'
import { TConnectConfigRequest, TCallbackConfigRequest, EAuthType } from './types'
import {
  getConfig
  // getSetupDetails
} from '../clients/integrations'
// import { mocked } from 'ts-jest/utils'
import { expandAuthConfig } from '../../api-config/auth-config'

jest.mock('../../clients/integrations')
jest.mock('../../api-config/auth-config')

describe('connectConfig', () => {
  const setup = (configureRequest: (req: TConnectConfigRequest) => void) =>
    new MiddlewareTestHarness({
      configureRequest,
      setupMiddlewares: [session()],
      testMiddleware: connectConfig
    })

  const setupProduction = (authType = 'OAUTH2') => {
    // const unexpandedIntegrationConfig = {
    //   authType,
    //   prodIntParam: '${some_var}'
    // }

    // const integrationConfig = {
    //   authType,
    //   prodIntParam: 'test-prod-int-value'
    // }

    // mocked(getConfig)
    //   .mockReturnValueOnce(unexpandedIntegrationConfig)
    //   .mockClear()
    // mocked(expandAuthConfig)
    //   .mockReturnValueOnce(integrationConfig)
    //   .mockClear()
    // mocked(getSetupDetails)
    // .mockImplementationOnce(() => ({ prodSetupParam: 'test-prod-setup-value' }))
    // .mockClear()

    return setup(req => {
      req.buid = 'test-buid'
      req.clientId = 'test-client-id'
      req.connectParams = { param: 'test-con-param' }
      req.setupId = 'test-setup-id'
    })
  }

  describe('when authenticating in production', () => {
    it('looks up the integration config and stores it on the request', async () => {
      const test = setupProduction()

      await test.get().expect(200)

      expect(getConfig).toMatchSnapshot()
      expect(test.req.integrationConfig).toEqual({
        authType: 'OAUTH2',
        prodIntParam: 'test-prod-int-value'
      })
    })

    it('expands any variables in the auth config', async () => {
      const test = setupProduction()

      await test.get().expect(200)

      expect(expandAuthConfig).toMatchSnapshot()
    })

    it('looks up the setup details and stores them on the request', async () => {
      const test = setupProduction()

      await test.get().expect(200)

      // expect(getSetupDetails).toMatchSnapshot()
      expect(test.req.setupDetails).toEqual({ prodSetupParam: 'test-prod-setup-value' })
    })

    it('saves the auth config into the session', async () => {
      const test = setupProduction()

      await test.get().expect(200)

      expect(test.req.session.authConfig).toMatchSnapshot()
    })
  })

  it('validates the authType', async () => {
    for (const validType of ['OAUTH1', 'OAUTH2', 'BASIC', 'APIKEY', 'NONE']) {
      await setupProduction(validType)
        .get()
        .expect(200)
    }

    const test = setupProduction('invalid')
    await test.get().expect(422)
    expect(test.err).toMatchSnapshot()
  })
})

describe('callbackConfig', () => {
  const setup = () =>
    new MiddlewareTestHarness<TCallbackConfigRequest>({
      configureRequest: req => {
        req.session!.authConfig = {
          integrationConfig: {
            authType: EAuthType.OAuth1,
            intParam: 'int-param-value'
          },
          setupDetails: {
            setupParam: 'setup-param-value'
          }
        }
      },
      setupMiddlewares: [session()],
      testMiddleware: callbackConfig
    })

  it('sets the auth config on the request, using the values stored in the session', async () => {
    const test = setup()

    await test.get().expect(200)

    expect(test.req.integrationConfig).toEqual({
      authType: EAuthType.OAuth1,
      intParam: 'int-param-value'
    })
    expect(test.req.setupDetails).toEqual({
      setupParam: 'setup-param-value'
    })
  })
})
