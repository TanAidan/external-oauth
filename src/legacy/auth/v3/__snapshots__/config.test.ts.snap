// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectConfig validates the authType 1`] = `[InvalidAuthType: Invalid authType 'invalid'. Please check the integration's auth.config.json file. Allowed values are 'OAUTH1', 'OAUTH2', 'BASIC', 'APIKEY', 'NONE']`;

exports[`connectConfig when authenticating in production expands any variables in the auth config 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "authConfig": Object {
          "authType": "OAUTH2",
          "prodIntParam": "\${some_var}",
        },
        "connectParams": Object {
          "param": "test-con-param",
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Object {
        "authType": "OAUTH2",
        "prodIntParam": "test-prod-int-value",
      },
    },
  ],
}
`;

exports[`connectConfig when authenticating in production looks up the integration config and stores it on the request 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "buid": "test-buid",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Object {
        "authType": "OAUTH2",
        "prodIntParam": "\${some_var}",
      },
    },
  ],
}
`;

exports[`connectConfig when authenticating in production looks up the setup details and stores them on the request 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "buid": "test-buid",
        "clientId": "test-client-id",
        "scopedUserDataTableName": "tableName",
        "setupId": "test-setup-id",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Object {
        "prodSetupParam": "test-prod-setup-value",
      },
    },
  ],
}
`;

exports[`connectConfig when authenticating in production saves the auth config into the session 1`] = `
Object {
  "integrationConfig": Object {
    "authType": "OAUTH2",
    "prodIntParam": "test-prod-int-value",
  },
  "setupDetails": Object {
    "clientId": "clientId",
    "clientSecret": "clientSecret",
  },
}
`;
